lldb/source/Plugins/ObjectFile/wasm/ObjectFileWasm.cpp

```diff
@@ -248,12 +248,93 @@ bool ObjectFileWasm::ParseHeader() {
   return true;
 }
 
-void ObjectFileWasm::ParseSymtab(Symtab &symtab) {}
+static std::vector<AddressRange> ParseFunctions(SectionSP code_section_sp) {
+  DataExtractor code_section_data;
+  code_section_sp->GetSectionData(code_section_data);
+  lldb::offset_t offset = 0;
+
+  const uint32_t function_count = code_section_data.GetULEB128(&offset);
+
+  std::vector<AddressRange> functions;
+  functions.reserve(function_count);
+
+  for (uint32_t i = 0; i < function_count; ++i) {
+    const uint32_t function_size = code_section_data.GetULEB128(&offset);
+    // llvm-objdump considers the ULEB with the function size to be part of the
+    // function. We can't do that here because that would break symbolic
+    // breakpoints, as that address is never executed.
+    functions.emplace_back(code_section_sp, offset, function_size);
+    offset += function_size;
```

Comment:
This should probably use an llvm addWithOverflow function to detect if we wrap around when reading garbage?: